### ISSUE TITLE
Support for 3rd party hardware

### ISSUE THREAD

**Issue by michaelsproul** [michaelsproul/xuino#2]
See: https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5---3rd-party-Hardware-specification

**Comment by prologic** [comment]
I actually happen to have TinyDunio(s) here that I'm working on with several proejcts. Is that what we consider 3rd-party? I'm going to be trying to get xuino to work with this hardware on my CRUX/LInux Desktop...

**Comment by michaelsproul** [comment]
Anything that requires a non-standard set of config values from boards.txt is third party.

I reckon we should copy the layout used by the Arduino IDE 1.5.x, so that people can easily switch to Xuino. The only problem is that the new system makes almost every aspect of compilation configurable, including the compiler!

We may be looking at something like:

```
CC = $(shell xuino get compiler --board $(BOARD))
```

Which I think is a little bit extreme.

On the other hand, support for intel and arm boards would be cool.

**Comment by prologic** [comment]
Can I make a suggestion? Do we have to use Makefile(s) at all? If not, can we either directly download (_or depend on some package that provides it_) the Arduino IDE libraries and relevant config and configuration files (_but without depending on the Arduino IDE itself and it's largish dependency tree_); and if we can do away with Makefiel(s) which I find inflexible in the first place adopt the codebase to use Fabric?

Basically what I'm thinking is... Build the necessary bits into Xuino such that someone need only:

``` #!bash
xuino build
xuino upload
```

Of sorts... :)

**Comment by michaelsproul** [comment]
What do you find inflexible about Makefiles? Part of my aim with this was to make Arduino development more C-like. I think it's useless to hide what's really going on behind a layer of abstraction, particularly if that layer is likely to break (which is the problem with Ino).

You're welcome to fork and prosper, but it's not my thing.

We also already have:

```
xuino init
xuino make
make upload
```

**Comment by prologic** [comment]
Cool okay :) It was just a random thought! I haven't fully tried xuino yet
(_trying to improve it's packaging!_)

James Mills / prologic

E: prologic@shortcircuit.net.au
W: prologic.shortcircuit.net.au
